encoder/vaapiencoder_base.cpp

```diff
@@ -287,7 +288,10 @@ SurfacePtr VaapiEncoderBase::createNewSurface(uint32_t fourcc)
     case VA_FOURCC_I420:
         rtFormat = VA_RT_FORMAT_YUV420;
         break;
-    case VA_FOURCC_YUY2:
+    case VA_FOURCC_P010:
+        rtFormat = VA_RT_FORMAT_YUV420_10BPP;
```

Comment:
this will build failed on libva 0.37. Could you use getRtFormat and check the return value?